.github/workflows/release.yml

```
@@ -1,17 +1,23 @@
-name: Release to Hackage
+name: Release
 
 on:
   push:
-    tags:
-      - 'v*'
+    branches: main
 
 jobs:
   release:
     runs-on: ubuntu-latest
-    env:
-      HACKAGE_API_KEY: ${{ secrets.HACKAGE_API_KEY }}
     steps:
       - uses: actions/checkout@v2
-      - uses: freckle/stack-upload-action@main
+
+      - id: tag
+        uses: freckle/haskell-tag-action@main
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - if: steps.tag.tag
```

I had a bug in the reference PR, which I commented on after the fact. Oops.

```suggestion
      - if: steps.tag.outputs.tag
```

---

Thanks!